addons/account/models/account.py

```diff
@@ -850,7 +850,7 @@ def write(self, vals):
                     bank_account = self.env['res.partner.bank'].browse(vals['bank_account_id'])
                     if bank_account.partner_id != company.partner_id:
                         raise UserError(_("The partners of the journal's company and the related bank account mismatch."))
-            if 'alias_name' in vals:
+            if 'alias_name' in vals or 'type' in vals and self.alias_id:
```

Comment:
I don't understand why it is done in `write` and not in `create`

Reply:
@william-andre yes indeed, you are right, we must add it in the create.

I thought we have all the fields in vals in the create function, as well as the "'alias_name" field. But this is not the case if we create via a server action.

Could you check the update?
Thanks!